boa/src/builtins/array/mod.rs

```diff
@@ -936,26 +948,30 @@ impl Array {
     /// [mdn]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/slice
     pub(crate) fn slice(this: &Value, args: &[Value], context: &mut Context) -> Result<Value> {
         let new_array = Self::new_array(context)?;
-        let len = this.get_field("length").as_number().unwrap() as i32;
+        let len = this.get_field("length").to_length(context)? as isize;
 
         let start = match args.get(0) {
-            Some(v) => v.as_number().unwrap() as i32,
-            None => 0,
+            Some(v) => v.to_integer(context)?,
+            None => 0.0,
         };
         let end = match args.get(1) {
-            Some(v) => v.as_number().unwrap() as i32,
-            None => len,
+            Some(v) => v.to_integer(context)?,
+            None => len as f64,
```

Comment:
This should use `f64::try_from()` I think.

Reply:
There is no `TryFrom<isize>` for `f64`, unfortunately. However, the cast should be safe (that is true for `isize == i32`, I'm not so sure for `isize == i64`).